src/hooks/useTokenBalance.ts

```diff
@@ -16,15 +16,19 @@ const useTokenBalance = (tokenAddress: string) => {
   const block = useBlock()
 
   const fetchBalance = useCallback(async () => {
+    if (!tokenAddress) {
+      return
+    }
+
     const balance = await getBalance(ethereum, tokenAddress, account)
     setBalance(new BigNumber(balance))
-  }, [account, ethereum, tokenAddress])
+  }, [account, ethereum, tokenAddress, setBalance])
```

Comment:
We can remove `setBalance` as it never changes.
```suggestion
  }, [account, ethereum, tokenAddress])
```